appcues/src/main/java/com/appcues/util/NotificationBuilderExt.kt

```diff
@@ -0,0 +1,105 @@
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import androidx.core.content.ContextCompat
+import com.appcues.AppcuesFirebaseMessagingService.AppcuesMessagingData
+import com.appcues.R
+import com.google.firebase.messaging.RemoteMessage
+import java.io.InputStream
+import java.net.HttpURLConnection
+import java.net.URL
+
+// for now channels are not configurable, which means all push messages will come through the default channel defined in XML config
+internal fun Context.getNotificationBuilder(): NotificationCompat.Builder {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+        with(getSystemService(Application.NOTIFICATION_SERVICE) as NotificationManager) {
+
+            createNotificationChannel(
+                NotificationChannel(
+                    getString(R.string.appcues_notification_channel_id),
+                    getString(R.string.appcues_notification_channel_name),
+                    resources.getInteger(R.integer.appcues_notification_channel_importance)
+                ).apply { description = getString(R.string.appcues_notification_channel_description) }
+            )
+        }
+    }
+
+    return NotificationCompat.Builder(this, getString(R.string.appcues_notification_channel_id))
+}
+
+internal fun NotificationCompat.Builder.setupNotification(message: RemoteMessage) = apply {
+    setAutoCancel(true)
+    setPriority(message.priority)
+    message.notification?.visibility?.let { setVisibility(it) }
+    message.notification?.channelId?.let { setChannelId(it) }
+}
+
+internal fun NotificationCompat.Builder.setContent(data: AppcuesMessagingData) = apply {
+    setContentTitle(data.title)
+    setContentText(data.body)
+}
+
+internal fun NotificationCompat.Builder.setStyle(context: Context, data: AppcuesMessagingData) = apply {
+    downloadImageFromUrl(data.image)?.let { image ->
+        setStyle(
+            NotificationCompat.BigPictureStyle()
+                .bigPicture(image)
+                .setSummaryText(data.body)
+        )
+    } ?: run {
+        setStyle(
+            NotificationCompat.BigTextStyle()
+                .bigText(data.body)
+        )
+    }
+
+    setSmallIcon(R.drawable.appcues_notification_small_icon)
+    setColor(ContextCompat.getColor(context, R.color.appcues_notification_color))
+}
+
+internal fun NotificationCompat.Builder.setIntent(context: Context, data: AppcuesMessagingData) = apply {
+    val intent = if (data.test) {
+        // during testing we just want to validate that push message came through
+        Intent(Intent.ACTION_VIEW, Uri.parse("appcues-${data.appId}://sdk/debugger/${data.id}"))
+    } else {
+        Intent(Intent.ACTION_VIEW, Uri.parse("appcues-${data.appId}://sdk/notification")).apply {
+            putExtra("notification_id", data.id)
+            putExtra("forward_deeplink", data.deeplink)
+            putExtra("show_content", data.experienceId)
+        }
+    }
```

Comment:
sorry but I think we may still have some details to work out here since the `test` value will be true on both the test push AND any non-published content like builder preview pushes.

I think we need to handle these scenarios
1. the notification ID matches what we had sent for a test - this is the case we mark the test successful in the debugger
2. the notification ID has `test: true` - this is some other preview type of push, it should still execute as normal with deep links, flow links, etc - for preview - but should NOT track analytics
3. all others - normal case - should execute any linked content AND track analytics

Reply:
its done now see if that makes sense